flake.nix

```
@@ -1,26 +1,44 @@
 # SPDX-FileCopyrightText: 2024 Sefa Eyeoglu <contact@scrumplex.net>
 #
 # SPDX-License-Identifier: MIT
-
 {
   description = "Nixpkgs overlay for bleeding-edge XR/AR/VR packages";
 
+  nixConfig = {
+    extra-substituters = ["https://cache.garnix.io"];
+    extra-trusted-public-keys = ["cache.garnix.io:CTFPyKSLcx5RMJKfLo5EEPUObbA78b0YQ2DTCJXqr9g="];
+  };
+
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
   };
 
-  outputs = inputs@{ flake-parts, nixpkgs, ... }:
-    flake-parts.lib.mkFlake { inherit inputs; } {
-      systems = [ "x86_64-linux" "aarch64-linux" ];
+  outputs = {
+    nixpkgs,
+    self,
+  }: let
+    systems = ["x86_64-linux" "aarch64-linux"];
+
+    forSystem = system: fn: fn nixpkgs.legacyPackages.${system};
+    forAllSystems = fn: nixpkgs.lib.genAttrs systems (system: forSystem system fn);
+  in {
+    devShells = forAllSystems (pkgs: {
+      default = pkgs.mkShell {
+        packages = with pkgs; [nvfetcher reuse];
+      };
+    });
+
+    formatter = forAllSystems (pkgs: pkgs.alejandra);
 
-      imports = [
-        ./pkgs
-      ];
+    packages = forAllSystems (pkgs: self.overlays.default pkgs pkgs);
 
-      perSystem = { config, self', inputs', pkgs, system, ... }: {
-        devShells.default = pkgs.mkShell {
-          packages = with pkgs; [nvfetcher reuse];
-        };
+    overlays.default = final: prev: let
+      sources = import ./_sources/generated.nix {
+        inherit (final) fetchgit fetchurl fetchFromGitHub dockerTools;
       };
-    };
+
+      mkOverride = pkg: newAttrs: prev.${pkg}.overrideAttrs (_: newAttrs);
+    in
+      prev.lib.mapAttrs mkOverride sources;
```

`prev` should exclusively be used for overrides

```suggestion
      final.lib.mapAttrs mkOverride sources;
```